### PR TITLE
bpo-39259: nntplib.NNTP/NNTP_SSL refactoring

### PR DESCRIPTION
**Pull request by corona10**
This PR is only refactoring work, so the news is skipped. :)

https://bugs.python.org/issue39259

### CODE REVIEW

**Comment by vstinner** [comment]
> This PR is only refactoring work, so I skip the news and issue.

I prefer to always attach a bpo to get more context. This change is connected to: https://bugs.python.org/issue39259

File: Lib/nntplib.py [hunk 1]
```diff
@@ -1057,15 +1055,21 @@ def __init__(self, host, port=NNTP_PORT, user=None, password=None,
             self.sock.close()
             raise
 
+    def _create_socket(self, timeout):
+        if timeout is not None and not timeout:
+            raise ValueError('Non-blocking socket (timeout=0) is not supported')
```
Comment (vstinner) [inline Lib/nntplib.py]: This is unrelated, please revert it.

**Review by vstinner — CHANGES_REQUESTED** [review]
Hum, you changed how _NNTPBase. With this change, I'm not sure that it's still relevant to have _NNTPBase *and* NNTP. What do you think of merging _NNTPBase and NNTP into NNTP? The most tricky part is to merge the two `__init__()` methods. Maybe just move NNTP code at the end of _NNTPBase.__init__().

File: Lib/nntplib.py (hunk 1 above)
Comment (corona10) [reply Lib/nntplib.py]: Oops Thanks for letting me know

**Comment by corona10** [comment]
@vstinner  Yes, that's reasonable. I will update this PR with merging 2 classes.

**Comment by corona10** [comment]
@vstinner 
https://github.com/python/cpython/blob/2e6a8efa837410327b593dc83c57492253b1201e/Lib/test/test_nntplib.py#L424

Hmm... _NNTPBase is also used for testing server code.
IMHO, if we merge _NNTPBase and NNTP, the work looks pretty large.
How about creating a separate issue for this (e.g remove _NNTPBase)?

File: Lib/nntplib.py [hunk 2]
```diff
@@ -1074,27 +1076,19 @@ def __init__(self, host, port=NNTP_SSL_PORT,
...
             """
-            sys.audit("nntplib.connect", self, host, port)
-            self.sock = socket.create_connection((host, port), timeout)
-            file = None
+            self.ssl_context = ssl_context
+            super().__init__(host, port, user, password, readermode,
+                             usenetrc, timeout)
+
+        def _create_socket(self, timeout):
+            self.sock = super()._create_socket(timeout)
```
Comment (vstinner) [inline Lib/nntplib.py]: In NNTP, the method only creates a socket, it doesn't assign self.sock. Please replace `self.sock` with `sock`.

**Comment by vstinner** [comment]
> Hmm... _NNTPBase is also used for testing server code.
> IMHO, if we merge _NNTPBase and NNTP, the work looks pretty large.
> How about creating a separate issue for this (e.g remove _NNTPBase)?

Ok.

**Comment by corona10** [comment]
@vstinner  Updated!

> Ok.

I will create a new issue for this :)

**Comment by corona10** [comment]
https://bugs.python.org/issue39305 is created :)

**Pull request merged**